### DISCUSSION TITLE
Issues installing ggplot

### DISCUSSION THREAD

**Discussion by crmateo** [RobertsLab/resources#1245] [Q&A] [answered]
Hi all! I'm running into an odd issue when installing ggplot2 in R studio. I had it previously installed but it said was not, however when I tried to re-install the package (install.packages("ggplot2")) I would get the following:

** byte-compile and prepare package for lazy loading
Error in loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]]) : 
  there is no package called ‘colorspace’
Calls: <Anonymous> ... loadNamespace -> withRestarts -> withOneRestart -> doWithOneRestart
Execution halted
ERROR: lazy loading failed for package ‘ggplot2’
* removing ‘/Library/Frameworks/R.framework/Versions/4.1/Resources/library/ggplot2’
The downloaded source packages are in
	‘/private/var/folders/c2/6158rqf16ns89f80w280gj3m0000gn/T/RtmpoWpCE3/downloaded_packages’
✔ Package 'ggplot2' successfully installed.
Warning message:
In utils::install.packages("ggplot2", repos = "https://cran.rstudio.com/") :
  installation of package ‘ggplot2’ had non-zero exit status

I still get warnings that ggplot2 needs to be installed to run the code. I tried installing tidyverse instead and loading the colorspace package mentioned in the error message (using install.packagess and lookup) with no success. I've I also tried updating R & R studio as well as updating my which didn't work.

I'm running RStudio ver. 1.4.1717 and R 4.1.0 GUI 1.76 High Sierra build (7976) on Mac OS Big Sur. I didn't think my big issue would be loading ggplot! :(

**Comment by kubu4** [comment]
Give this a whirl to try installing `colorspace` and report back:

```R
install.packages("https://cran.r-project.org/bin/macosx/contrib/4.1/colorspace_2.0-1.tgz", repos=NULL)
```

We'll try to tackle this first and then see what we can do about `ggplot2`.

EDITED: Improved markdown formatting.

**Reply by crmateo** [reply]
Here's the result:

> trying URL 'https://cran.r-project.org/bin/macosx/contrib/4.1/colorspace_2.0-1.tgz'
Warning in install.packages :
  cannot open URL 'https://cran.r-project.org/bin/macosx/contrib/4.1/colorspace_2.0-1.tgz': HTTP status was '404 Not Found'
Error in download.file(p, destfile, method, mode = "wb", ...) : 
  cannot open URL 'https://cran.r-project.org/bin/macosx/contrib/4.1/colorspace_2.0-1.tgz'

**Comment by kubu4** [comment, marked as answer]
```R
install.packages("https://cran.r-project.org/bin/macosx/contrib/4.1/colorspace_2.0-2.tgz", repos=NULL)
```

Okay, try the command above. I've verified that this link works, so you shouldn't get the 404 error this time. However, who knows what else will happen. :)

**Reply by crmateo** [reply]
Thanks so much for thee solution! It's working as expected for once! :)
